iceoryx_hoofs/vocabulary/include/iox/detail/optional.inl

```diff
@@ -306,7 +306,10 @@ bool operator==(const optional<T>& lhs, const optional<T>& rhs) noexcept
 template <typename T>
 bool operator!=(const optional<T>& lhs, const optional<T>& rhs) noexcept
 {
-    return !(lhs == rhs);
+    const auto onlyLhsNul = !lhs.has_value() && rhs.has_value();
+    const auto onlyRhsNul = lhs.has_value() && !rhs.has_value();
```

Comment:
Why not `onlyLhsNull` and `onlyRhsNull`?

Reply:
When one of the values is null and the other is not they are not equal but if they both contain a value we have to delegate this to the inequality operator of the type.

Does it help?!

Reply:
@elfenpiff I was referring to the names. I wondered why there is no second `l` in `*Nul`. Sorry, I should have phrased that more clearly.

Reply:
@FerdinandSpitzschnueffler ...mmmhhh I think I am lazy ;)